### ISSUE TITLE
3.0.2 - Unexpected Results in Camera transforms not present in 3.0.1 (C#)

### ISSUE THREAD

**Issue by nazgulsenpai** [godotengine/godot#17306]
**Godot version:**
3.0.2 mono

**OS/device including version:**
GTX 1050 (tested also on GTX 950M)

**Issue description:**
The Camera/PlayerMovement scripts I wrote in 3.0.1 moves in an unexpected manner in 3.0.2. The exact same project produces different results. I tested in 3.0 (Windows) also and it works as it should, so either this is a regression or a change that I didn't find in documentation. I have tested this on Windows, MacOS High Sierra, and Linux (Ubuntu 17.10) builds (of 3.0.1 and 3.0.2, respectively) before submitting the bug and the problem persists across platforms.

The camera rotates oddly, and the orientation of the player movement becomes unpredictable as movement is based on camera.transform.basis

The reproduction project is 25MB (as low as I could get it to trim without rewriting it) so I will provide a GoogleDrive link of the whole project and the ZIP attached just includes the CS scripts.

The scene is setup like this:
![scene_tree](https://user-images.githubusercontent.com/31775906/37041225-4e8874d8-2121-11e8-8dd9-9f4b1480e0b8.png)

**Steps to reproduce:**
Attach PlayerMovement.cs to a Kinematic Body. Add a camera as a child of KinematicBody and attach the FollowCamera.cs script

3.0.1:
![problem-godot3 0 1](https://user-images.githubusercontent.com/31775906/37040919-8d2a94ba-2120-11e8-82bd-5911ea572bc9.gif)
3.0.2:
![problem-godot3 0 2](https://user-images.githubusercontent.com/31775906/37040939-9841fca8-2120-11e8-9b4e-c5eb4bbdd8f4.gif)

**Minimal reproduction project:**
https://drive.google.com/file/d/16L-DgzgNm6x0laPMDS_HNvpNyshT56OX/

**Comment by mysticfall** [comment]
Maybe the same issue as #16937?

**Comment by nazgulsenpai** [comment]
I think so. I'll build the latest master and give it a go. Thank you!

**Comment by mysticfall** [comment]
@nazgulsenpai Just to make it clear, if you want to confirm whether or not it's caused by the same issue as mine, you'll probably need to try either the official 3.0.2 build or build `dad47d8876de45132226985dbbd079e060f6facc` rather than the latest master branch.

**Comment by hpvb** [comment]
The initial set of patches for those functions were removed from 3.0.2 because they caused unrelated problems. If it behaves differently from 3.0 then it appears that I've not reverted all relevant parts of these fixes. I'll have a look.

**Comment by akien-mga** [comment]
The bug report says that it behaves differently from 3.0.1, which is the version where the behaviour was deemed unsatisfactory and thus reverted back to the 3.0 state in 3.0.2.

So this change is expected, and the 3.0.2 behaviour is (currently) the expected one.

**Comment by Sslaxx** [comment]
Except the initial report here states 3.0.0 also had the behaviour that they expected.

**Comment by mysticfall** [comment]
Yeah, as we currently understand it, if something works 'correctly' in 3.0.1 but not in 3.0.2, then it shouldn't work as expected in 3.0 either.

So, if we can reproduce the issue with 3.0.2 but not with dad47d8 or 3.0, then probably we better check if we have rolled back all the releant changes correctly as @hpvb mentioned.

But if we find that the example project only works correctly in 3.0.1, then probably the supposed 'correct' behavior is in fact the opposite, so we can safely close this issue as a duplicate of #16937.

**Comment by nazgulsenpai** [comment]
I'm sorry - the bug DOES exist in 3.0-stable_mono_win64. I don't know how I managed to confuse myself but I have. 

So to clear it up, the problem existed in 3.0.0 and 3.0.2. 

I just tested with a fresh pull/build from master branch (about 5 hours ago) and it appears the transform issue is fixed, but IDK where to even start here.

![ezgif-5-2a2c8e1f4a](https://user-images.githubusercontent.com/31775906/37092273-d4f2e4cc-21d0-11e8-86de-f0e60f2e1abf.gif)

**Comment by nazgulsenpai** [comment]
Ok, great. Thanks for the heads up and I'll test as soon as I get to the office. 

Thank you guys so much for all of your assistance
